ulib/experimental/Steel.SelEffect.fsti

```diff
@@ -257,6 +272,9 @@ val change_slprop (p q:vprop) (vp:erased (normal (t_of p))) (vq:erased (normal (
     (ensures interp (hp_of q) m /\ sel_of q m == reveal vq)
   ) : SteelSel unit p (fun _ -> q) (fun h -> h p == reveal vp) (fun _ _ h1 -> h1 q == reveal vq)
 
+val change_equal_slprop (p q: vprop)
+  : SteelSel unit p (fun _ -> q) (fun _ -> p == q) (fun h0 _ h1 -> p == q /\ h1 q == h0 p)
```

Comment:
You might want to express the postcondition with a normalized frame_equalities (see get for instance). Else, if p/q are not atomic subresources, you will lose equalities

Reply:
I intend this `change_equal_slprop` to be a specialized version of `change_slprop` where the user "provides" an equality proof rather than a proof involving memory states, which is why I made the shape of their pre and postconditions similar.

Reply:
Ah, I can see the difference. With `change_slprop` the user provides ghost values "as a whole" and there is no intent to prove more fine-grained equalities than with those ghost values. But with `change_equal_slprop` if `p` or `q` is not atomic, then the granularity may be in the definitions of `p` or `q`. Thank you Aymeric for your comment.
However, in fact, I do not intend to rely on normalization here, not just because `p` or `q` might most often be atomic, but also if the equality proof `p == q` is not a proof by normalization.

Reply:
I was mostly thinking of the `h1 q == h0 p` postcondition here. If p is, say, `p1 * p2`, then you will not obtain equalities on p1 and p2, and you will need to manually call a lemma to obtain them from an equality on p1 * p2;
The convention we've been following is that everything is expressed on atomic vprops, since composite resources will be reshuffled often.
The alternative I'm proposing is to restate this as a `frame_equalities p h0 h1` (with a call to normal to reduce this recursive predicate at the call site).